source/slang/slang-compiler.h

```diff
@@ -1408,6 +1410,8 @@ namespace Slang
         /// Set if fileSystemExt is a cache file system
         RefPtr<CacheFileSystem> m_cacheFileSystem;
 
+        LineDirectiveMode m_lineDirectiveMode = LineDirectiveMode::Default;
```

Comment:
Okay, I'm clearly missing something here. Why does the linkage/session need a line-directive mode stored on it? It really belongs on the targets, AFAICT. Or does the existing logic not correctly associate line-directive modes with specific targets?